cmd/dp-dimension-importer/main.go

```diff
@@ -58,28 +81,61 @@ func main() {
 		return repository.NewDimensionRepository(neo4jClient, map[string]string{})
 	}
 
-	importAPI := client.ImportAPI{
-		ResponseBodyReader: responseBodyReader{},
-		HTTPClient:         &http.Client{},
-		AuthToken:          cfg.ImportAuthToken,
-		ImportHost:         cfg.ImportAddr,
-	}
+	// ImportAPI HTTP client.
+	importAPI := client.NewImportAPI(cfg.ImportAddr, cfg.ImportAuthToken, responseBodyReader{}, &http.Client{})
 
-	eventHandler := &handler.DimensionsExtractedEventHandler{
-		NewDimensionInserter: newDimensionInserterFunc,
-		InstanceRepository:   &repository.InstanceRepository{Neo4j: neo4jClient},
-		ImportAPI:            importAPI,
-	}
+	// Handler for dimensionsExtracted events.
+	eventHandler := handler.NewDimensionExtractedEventHandler(
+		newDimensionInserterFunc,
+		&repository.InstanceRepository{Neo4j: neo4jClient},
+		importAPI)
+
+	// MessageProducer for dimensionsInsertedEvents
+	instanceCompletedProducer := message.NewInstanceCompletedProducer(
+		outgoingInstances,
+		schema.InstanceCompletedSchema,
+	)
 
-	dimensionInsertedProducer := message.DimensionInsertedProducer{
-		Producer:   insertedEventProducer,
-		Marshaller: schema.DimensionsInsertedSchema,
+	// Errors handler
+	errorEventHandler := &handler.ErrorHandler{
+		Producer:   errorEventProducer,
+		Marshaller: schema.ErrorEventSchema,
 	}
 
-	exitChannel := make(chan error)
-	err = message.Consume(extractedEventConsumer, dimensionInsertedProducer, eventHandler, exitChannel)
-	if err != nil {
-		log.ErrorC("consumer", err, nil)
+	// HTTP Health check endpoint.
+	healthcheck.NewHandler(cfg.BindAddr)
+
+	gracefulShutdown := func() {
+		log.Info(gracefulShutdownMsg, nil)
+		ctx, _ := context.WithTimeout(context.Background(), time.Second*cfg.ShutdownTimeout)
```

Comment:
Do I need to multiply by seconds here if the return is already a duration?

Reply:
If its already a duration you can just use it as is. Multiplying by seconds will make the value incorrect.